inference-engine/tests/functional/shared_test_classes/src/subgraph/transpose_add.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/subgraph/transpose_add.hpp"
+
+namespace SubgraphTestsDefinitions {
+std::string TransposeAdd::getTestCaseName(testing::TestParamInfo<TransposeAddParams> obj) {
+    InferenceEngine::Precision netPrecision;
+    std::string targetName;
+    std::vector<size_t> input_shape;
+    std::map<std::string, std::string> configuration;
+
+
+    std::tie(netPrecision, targetName, input_shape, configuration) = obj.param;
+    std::ostringstream results;
+
+    results << "IS=" << CommonTestUtils::vec2str(std::vector<size_t>(input_shape.begin(), input_shape.end())) << "_";
+    results << "netPRC=" << netPrecision.name() << "_";
+    results << "targetDevice=" << targetName;
+    return results.str();
+}
+
+void TransposeAdd::SetUp() {
+    InferenceEngine::Precision netPrecision;
+    std::vector<size_t> input_shape;
+    std::map<std::string, std::string> additional_config;
+
+    std::tie(netPrecision, targetDevice, input_shape, additional_config) = this->GetParam();
+
+    configuration.insert(additional_config.begin(), additional_config.end());
+
+    auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+    auto params = ngraph::builder::makeParams(ngPrc, {input_shape});
+
+    ngraph::Shape permute_order(input_shape.size());
+    std::iota(std::begin(permute_order), std::end(permute_order), 0);
+    std::iter_swap(std::end(permute_order) - 2, std::end(permute_order) - 1);
```

Comment:
consider to check whether `input_shape.size()  >= 2`

Reply:
check is added